CONTRIBUTING.rst

```diff
@@ -0,0 +1,162 @@
+Contributing to the Jupyter Notebook
+====================================
+
+TODO: a welcoming sentence
+
+General Guidelines
+------------------
+
+For general documentation about contributing to Jupyter projects, see the
+`Project Jupyter Contributor Documentation`__.
+
+__ http://jupyter.readthedocs.org/#contributor-documentation
+
+
+Setting Up a Development Environment
+------------------------------------
+
+If you have already installed the dependencies mentioned below, the following
+steps should get you going::
+
+    pip install setuptools pip --upgrade --user
+    git clone https://github.com/jupyter/notebook
+    cd notebook
+    pip install -e . --user
+
+If you want this to be available for all users of your system (assuming you
+have the necessary rights) or if you are running the commands in a virtualenv,
+just drop the ``--user`` option.
+
+Installing the Dependencies
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Python Development Libraries
+""""""""""""""""""""""""""""
+
+On Debian/Ubuntu systems, you can get them with::
+
+    sudo apt-get update
+    sudo apt-get install python3-dev
+
+The development libraries might be needed for the installation of *PyZMQ*,
+*Tornado* and *Jinja2*.
+
+Alternatively -- if you prefer -- you can also install those packages directly
```

Comment:
I actually tried this (see https://github.com/jupyter/notebook/issues/1193#issuecomment-196798100). I'll come back to that later ...